Acquire GIL before creating py::object in RPC python handler

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26988 Acquire GIL before creating py::object in RPC python handler**

Differential Revision: [D17635297](https://our.internmc.facebook.com/intern/diff/D17635297)

mrshenli committed:
Acquire GIL before creating py::object in RPC python handler

zhaojuanmao commented on torch/csrc/distributed/rpc/python_functions.cpp:
```diff
@@ -53,7 +53,12 @@ py::object toPyObj(const Message& message) {
+      {
+        AutoGIL ag;
+        // createPyObjectForStack does not acquire GIL but creates a new
+        // py::object
```
nit: but creates a new py::object needs GIL

mrshenli committed:
Update on "Acquire GIL before creating py::object in RPC python handler"


Differential Revision: [D17635297](https://our.internmc.facebook.com/intern/diff/D17635297)
